### PR TITLE
support assigning to a mutable reference in cube macro

### PR DESCRIPTION
**Pull request by cBournhonesque**

### CODE REVIEW

File: crates/cubecl-macros/src/codegen_function/variable.rs [hunk 1]
```diff
@@ -167,6 +167,18 @@ pub(crate) fn codegen_assign(
...
+            let rhs = codegen_expr(&assign.right, loop_level, variable_tracker);
+
+            quote::quote! {
+                {
+                    let _assign_lhs = #lhs;
+                    let _assign_rhs = #rhs;
+                    cubecl::frontend::assign::expand(context, _assign_rhs, _assign_lhs)
+                }
+            }
+        }
```
Comment (nathanielsimard) [inline crates/cubecl-macros/src/codegen_function/variable.rs:181]: Could we use the | operator for this?

```rust
 syn::Expr::Unary(_) | syn::Expr::Field(_) | sync::Expr::Path(_) => {
      let lhs = codegen_expr(&assign.left, loop_level, variable_tracker);
      let rhs = codegen_expr(&assign.right, loop_level, variable_tracker);

      quote::quote! {
          {
              let _assign_lhs = #lhs;
              let _assign_rhs = #rhs;
              cubecl::frontend::assign::expand(context, _assign_rhs, _assign_lhs)
          }
      }
  }
```

**Review by nathanielsimard — COMMENTED** [review]
Maybe we can reduce code duplication a bit?

File: crates/cubecl-macros/src/codegen_function/variable.rs (hunk 1 above)
Comment (cBournhonesque) [reply crates/cubecl-macros/src/codegen_function/variable.rs]: good call

**Comment by cBournhonesque** [comment]
The format checks fail in ci but not locally

**Pull request merged**